### DISCUSSION TITLE
Pre-filled data

### DISCUSSION THREAD

**Discussion by Tankers-balloon** [sqlitebrowser/sqlitebrowser#3413] [Q&A] [answered]
Hi !

I am new using DB Browser for SQLite and I would like to provide a scrolling list of values to help the database user fill a column. Is it possible to do this using DB Browser ?
I did not find any solution until now...

Thanks a lot for your help !

**Comment by chrisjlocke** [comment]
This is using foreign keys, and while DB4S provides the IDs to use (and only allows the correct IDs) it doesn't display what those IDs belong to.

**Reply by Tankers-balloon** [reply]
Thanks for your help ! Since I am a beginner in databases I am not sure I fully understand your answer, despite the documentation of SQLite...
As far as I understand, prefill column requires to store the data ni another database and make a link using foreign key ? And since DB4S provides it own IDs, it is not possible to do so ?

**Reply by chrisjlocke** [reply, marked as answer]
Apologies for the short answer.
Linked data would be in a different table, not necessarily in a different database.  It can be in another database, but more common to keep it in one database.
So if you had a list of people, and wanted to link them to the country they lived, you'd have two tables.

```
people                                           countries
id       | name    | countryId                   id     | country
--------------------------------                 ---------------------------------
1        | jack    |                             1      | UK
2        | alan    |                             2      | Spain
3        | dave    |                             3      | Italy
```
If Jack was in Spain, you'd have to assign **2** to his countryId field.  If Alan was in the UK, he would have **1**, etc.  So you have to know what ID you wanted to assign.  DB4S can't give you a dropdown of 'UK, Spain, Italy' and assign it '1, 2, 3' automatically.  Its a common 'feature' of other databases, and a common requirement.  Just not something we've gotten around to implementing.  We have a sort-of stab of it, but it just displays the IDs, not the related field.

**Reply by Tankers-balloon** [reply]
Thanks a lot @chrisjlocke for these details, now I get it :)
